sympy/functions/special/holonomic.py

```diff
@@ -0,0 +1,428 @@
+
+    One can use the `Dx` and represent differential operators more conveniently this way also.
+    >>> (x*Dx*x + 1 - Dx**2)**2
+    (2*x**2 + 2*x + 1) + (4*x**3 + 2*x**2 - 4)Dx + (x**4 - 6*x - 2)Dx**2 + (-2*x**2)Dx**3 + (1)Dx**4
+
+    """
+
+    _op_priority = 30
+
+    def __init__(self, list_of_poly, parent, generator=None, var=None):
+
+        if generator is None:
+            self.gen_symbol = symbols('Dx', commutative=False)
+        else:
+            if isinstance(generator, str):
+                self.gen_symbol = symbols(generator, commutative=False)
+            elif isinstance(generator, Symbol):
+                self.gen_symbol = generator
+
+        self.parent = parent
+
+        if isinstance(list_of_poly, list):
+            for i, j in enumerate(list_of_poly):
+                if isinstance(j, int):
+                    list_of_poly[i] = (
+                        (self.parent).base).from_sympy(S(j))
+                elif not isinstance(j, self.parent.base.dtype):
+                    list_of_poly[i] = (
+                        (self.parent).base).from_sympy(j)
+
+            self.listofpoly = list_of_poly
+
+    def args(self):
+
+        return self.gen_symbol
+
+    def __mul__(self, other):
+        """
+        Multiplies two DifferentialOperator and returns another
+        DifferentialOperator isinstance using the commutation rule
+        Dx*a = a*Dx + a'
+        """
+
+        ring_0 = self.parent.base.zero
+        ring_1 = self.parent.base.one
+
+        if isinstance(other, DifferentialOperator):
+            if other.listofpoly == [ring_0, ring_1]:
+                sol = []
+                sol.append(ring_0)
+                for i in self.listofpoly:
+                    sol.append(i)
+
+                return DifferentialOperator(sol, self.parent, self.gen_symbol)
+
+        gen = DifferentialOperator([ring_0, ring_1], self.parent, self.gen_symbol)
+        listofpoly = self.listofpoly
+        sol = (listofpoly[0] * other)
+
+        def _diff_n_times(b):
+            expr = b * gen + b.diff()
+            return expr
+
+        for i in range(1, len(listofpoly)):
+            other = _diff_n_times(other)
+            sol += (listofpoly[i] * other)
+
+        return sol
+
+    def __rmul__(self, other):
+
+        if not isinstance(other, DifferentialOperator):
+
+            if isinstance(other, int):
+                other = S(other)
+
+            if not isinstance(other, self.parent.base.dtype):
+                other = (self.parent.base).from_sympy(other)
+
+            sol = []
+            for j in self.listofpoly:
+                sol.append(other * j)
+
+            return DifferentialOperator(sol, self.parent, self.gen_symbol)
+
+    def __add__(self, other):
+
+        if isinstance(other, DifferentialOperator):
+
+            list_self = self.listofpoly
+            list_other = other.listofpoly
+
+            if min(len(list_self), len(list_other)) is len(list_self):
```

Comment:
This could be ``if len(list_self) <= len(list_other):``

And the two constructions could follow immediately. There is no need to set the ``minimum`` flag.